### PR TITLE
feat: toc intersection observer

### PR DESCRIPTION
**Pull request by benediktgrether**

### CODE REVIEW

**Comment by sebald** [comment]
@benediktgrether awesome job 🥳 I have one suggestion to make it even more awesome: if the user scrolled all the way down to the buttom, can we make sure the last section is always highlighted? (e.g. "if scroll === bottom of page -> select last item)

Screen Shot 2022-09-06 at 13 27 09

**Comment by benediktgrether** [comment]
@sebald can we have a look tomorrow on this task? Because i found two issues.

**Review by sebald — CHANGES_REQUESTED** [review]
I found a little thing 🙈 there is no section highlighted initially :D

File: docs/src/components/Toc.tsx [hunk 1]
```diff
@@ -2,25 +2,90 @@ import { useEffect, useRef, useState } from 'react';
...
+      ) as any;
+      const elementsLenght = elements.length;
+      const activeElement =
+        elements[elementsLenght - 2]?.dataset.intersectionActive;
+      if (
+        window.innerHeight + window.scrollY >= document.body.scrollHeight &&
+        activeElement === 'true' &&
+        isBottom.current === false
+      ) {
+        elements[elementsLenght - 2].style.fontWeight = '400';
```
Comment (sebald) [inline docs/src/components/Toc.tsx:74]: Also: can we add a class or `[data-attribute]` and do the styling inside Toc!? :)

**Review by sebald — APPROVED** [review]

**Pull request merged**